internal/worker/worker.go

```diff
@@ -0,0 +1,84 @@
+package worker
+
+type Worker interface {
+	Registry
+	Start() error
+	// Run is a blocking start and cleans up resources when killed
+	// returns error only if it fails to start the worker
+	Run() error
+	// Stop cleans up any resources opened by worker
+	Stop()
+}
+
+type Registry interface {
+	RegisterWorkflow(w interface{})
+	RegisterWorkflowWithOptions(w interface{}, options RegisterWorkflowOptions)
+	RegisterActivity(a interface{})
+	RegisterActivityWithOptions(a interface{}, options RegisterActivityOptions)
+}
+
+var backendWorker Registry
```

Comment:
Shouldn't this be of type `Worker`? Also let's avoid global variables if possible. These can be initialized explicitly during starlark-worker process startup and passed around